config/ccConfig.js

```diff
@@ -1,5 +1,31 @@
 // The credentials for NYPL's Simplified Card Creator API
+require('dotenv').config();
+
+const aws = require('aws-sdk');
+let decryptKMS;
+let kms;
+
+kms = new aws.KMS({
+  region: 'us-east-1',
+});
+
+decryptKMS = (key) => {
+  const params = {
+    CiphertextBlob: new Buffer(key, 'base64'),
+  };
+
+  return new Promise((resolve, reject) => {
+    kms.decrypt(params, (err, data) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(data.Plaintext.toString());
+      }
+    });
+  });
+};
+
 module.exports = {
-  username: process.env.CARD_CREATOR_USERNAME,
-  password: process.env.CARD_CREATOR_PASSWORD
+  username: decryptKMS(process.env.CARD_CREATOR_USERNAME),
```

Comment:
Does `dotenv.config` assume a `.env` by default? Is one going to be available to travis?

Also `module.exports.username` is assigned `Promise` above - not a string value. The time we seem to use it is here: https://github.com/NYPL/dgx-patron-creator-service/blob/0758411d5c49a845bb49a21d3a017d462a0f4c62/api/controllers/createPatron.js#L113 Essentially we'll be sending `axios` a hash with `auth: { username: <Promise>, password: <Promise> }, ...`, right? I may be misreading something.

Reply:
That's right. It needs to be decrypted first before assigning the values to `username` and `password` and in this case it's returning a promise instead of the string values.

@LucasCioffi perhaps the kms decryption code should only go here, and in `createPatron` you call that decryption call and create this patron object there. You might want to call it directly before the axios call so that when the decryption is complete, then you can call axios on those values.